web/src/containers/SaveButton.js

```diff
@@ -25,87 +25,76 @@ const getButtonContent = (dirty, working) => {
   return 'Save';
 };
 
-class SaveButton extends PureComponent {
-  state = {
-    hasFetched: false,
-    success: false
-  };
+const SaveButton = ({ error, needsSave, saveApd: action, working }) => {
+  const [hasFetched, setHasFetched] = useState(false);
 
-  static getDerivedStateFromProps({ error, working }, { hasFetched }) {
-    // Success has to be derived.  It can't be stored in the app state because
-    // if it was, then the next time this form was loaded, it would show the
-    // success state even though it wouldn't be accurate anymore.
-    if (hasFetched) {
-      const success = !working && !error;
-      return { success };
-    }
-    return null;
-  }
+  // Success has to be derived.  It can't be stored in the app state because
+  // if it was, then the next time this form was loaded, it would show the
+  // success state even though it wouldn't be accurate anymore.
```

Comment:
One cool thing is that with functional components, you don't need pre-render lifecycle methods like `getDerivedStateFromProps`. You can just... derive the state from the props, and it's much clearer what's going on.

Reply:
The functional components are so much easier to follow in general -- I'm really excited about this PR!